src/devices/src/virtio/block/device.rs

```diff
@@ -270,6 +270,10 @@ impl VirtioDevice for Block {
         &mut self.queues
     }
 
+    fn queues_clone(&self) -> Vec<Queue> {
+        self.queues.clone()
+    }
```

Comment:
This is kinda unnecessary. Users of the `VirtioDevice` can always do
```rust
let cloned_queues = virtio_dev.queues().to_vec();
```
instead of exposing a new trait function to explicitly clone the queues.

Reply:
I'll remove it if it seems like VirtioDevice trait is not the place for it, but I included the method because it simplifies saving the state (having access to all these methods means we can directly implement `save` here and not have duplicate code in the devices).

Reply:
Initially I tried that, but the problem is that `queues` takes a mutable reference to `self` and this method doesn't (and shouldn't if we want to call it from the `Persist` `save` context).

Reply:
Yes, I see now. Seems like we do need to modify `trait VirtioDevice` in one way or another.

It is better to rename the existing `queues()` to `queues_mut()` since it provides a mut reference, and add a immutable getter for queues that doesn't clone, just provides `&[queue]`.

The reason for that, extracted from [this well-written answer](https://stackoverflow.com/a/35391084):

> Secondly, you should not clone in a getter. This is hugely wasteful if all the user wants to do is, for example, read from the value. It's better to return an immutable borrow, from which the user can clone if they need to.

Reply:
Done.